example/main.dart

```diff
@@ -28,14 +28,14 @@ Future main() async {
         ((event.client.x - boundingRect.left) + target.scrollLeft).floor();
     var endY = ((event.client.y - boundingRect.top) + target.scrollTop).floor();
 
-    var rectangle = Rectangle(startX, startY, endX - startX, endY - startY);
+    var rectangle = Rectangle(startX!, startY!, endX - startX!, endY - startY!);
 
     if (currentBrush == 'search') {
-      var resultList = querySelector('#results');
+      var resultList = querySelector('#results')!;
       resultList.children = [];
-      for (RTreeDatum match in rtree.search(rectangle)) {
+      for (RTreeDatum? match in rtree.search(rectangle)) {
```

Comment:
Type can be inferred.
```suggestion
      for (final match in rtree.search(rectangle)) {
```